lib/segment/src/index/field_index/uuid_index.rs

```diff
@@ -0,0 +1,182 @@
+use std::str::FromStr;
+use std::sync::Arc;
+
+use common::types::PointOffsetType;
+use delegate::delegate;
+use parking_lot::RwLock;
+use rocksdb::DB;
+use serde_json::Value;
+
+use super::numeric_index::NumericIndexInner;
+use super::{
+    CardinalityEstimation, FieldIndexBuilderTrait, PayloadBlockCondition, PayloadFieldIndex,
+    ValueIndexer,
+};
+use crate::common::operation_error::{OperationError, OperationResult};
+use crate::common::Flusher;
+use crate::index::field_index::histogram::Numericable;
+use crate::index::field_index::PrimaryCondition;
+use crate::telemetry::PayloadIndexTelemetry;
+use crate::types::{
+    FieldCondition, Match, MatchValue, PayloadKeyType, UuidPayloadKeyType, UuidPayloadType,
+    ValueVariants,
+};
+
+pub struct UuidPayloadIndex {
```

Comment:
Are you sure that we need separate structure for uuid? You can use `NumericIndex<u128, UuidPayloadKeyType>` and don't copy builder, delegations etc

Reply:
The reason I created a separate structure is because [here](https://github.com/qdrant/qdrant/pull/4738/files#diff-3fd7336e71c9c53184049c40a48f91408e1dd7ad63aa49ee2cf91d56d0b1ace7R82) and in estimation we try to cast a keyword into a UUID, which makes sense for UUID index but not for numeric index. 
I thought it would be cleaner to separate this behavior, especially because if a user accidentally passes a valid UUID to a number index, it would be parsed and the key would be compared to the numbers in the number index which doesn't make sense at all.

Reply:
We have already such behaviour (conversion from string) for datetime index:
https://github.com/qdrant/qdrant/blob/dev/lib/segment/src/index/field_index/numeric_index/mod.rs#L563
It seems you may only implement `impl ValueIndexer for NumericIndex<u128, UuidPayloadKeyType>` instead of new entity